tests/Jira/ApiTest.php

```diff
@@ -149,6 +149,40 @@ public function testFindVersionByName()
 		);
 	}
 
+	public function testGetResolutions()
+	{
+		$response = '[{"self":"https://test.atlassian.net/rest/api/2/resolution/1","id":"1","description":"A fix for this issue is checked into the tree and tested.","name":"Fixed"},{"self":"https://test.atlassian.net/rest/api/2/resolution/10000","id":"10000","description":"This issue won\'t be actioned.","name":"Won\'t Do"}]';
```

Comment:
Think I found a nice compromis between our solutions. Keeps the tests nice and short, no duplication but also the raw JSON to your disposal if needed:

		$response = file_get_contents(__DIR__.'/resources/api_status.json');

		$this->expectClientCall(
			Api::REQUEST_GET,
			'/rest/api/2/resolution',
			array(),
			$response
		);

		$actual = $this->api->getResolutions();

		$response_decoded = json_decode($response, true);

		$expected = [
			'1' => $response_decoded[0],
			'3' => $response_decoded[1],
		];
		$this->assertEquals($expected, $actual);

Also writing the same tests for getFields, getStatuses, getPriorities